src/main/java/cn/nukkit/player/handler/LoginPacketHandler.java

```diff
@@ -67,32 +71,15 @@ public boolean handle(LoginPacket packet) {
         }
 
         String username = this.loginData.getChainData().getUsername();
-        boolean valid = true;
-        int len = username.length();
-        if (len > 16 || len < 3) {
-            valid = false;
-        }
-
-        for (int i = 0; i < len && valid; i++) {
-            char c = username.charAt(i);
-            if ((c >= 'a' && c <= 'z') ||
-                    (c >= 'A' && c <= 'Z') ||
-                    (c >= '0' && c <= '9') ||
-                    c == '_' || c == ' '
-            ) {
-                continue;
-            }
+        Matcher matcher = pattern.matcher(username);
 
-            valid = false;
-            break;
-        }
-
-        loginData.setName(TextFormat.clean(username));
-        if (!valid || Objects.equals(loginData.getName().toLowerCase(), "rcon") || Objects.equals(loginData.getName().toLowerCase(), "console")) {
+        if (!matcher.matches() || username.matches("^(((R|r)(C|c)(O|o)(N|n))|((C|c)(O|o)(N|n)(S|s)(O|o)(L|l)(E|e)))$")) {
```

Comment:
This should be a pattern as well. Then you can use the case insensitive flag.

Reply:
I didn’t understand what you mean.

Reply:
why are you using a regex for that at all? you're simply doing a case-insensitive check against 2 strings, so just use `"rcon".equalsIgnoreCase(username) || "console".equalsIgnoreCase(username)`